main.js

```diff
@@ -10,3 +10,35 @@ document.querySelectorAll(".nav-link").forEach((n) =>
     sideMenu.classList.remove("active");
   })
 );
+
+// create website data preservation 
+
+// object containing form data
+const formInputData = {
+  nameInputData: '',
+  emailInputData: '',
+  messageInputData: ''
+}
+
+// check the availability of form data
+if (localStorage.getItem('formInputData')!==null){
+  // retrieve the data if it is there
+  const inputData = localStorage.getItem('formInputData');
+  // deserialize the data
+  formInputData = JSON.parse('inpuData');
+}
+
+// obtain the input and message data
+const nameEmailMessageData = document.querySelectorAll('input, textarea');
+// loop through the fields
+nameEmailMessageData.forEach((field) => {
+  field.value = formInputData[field.name];
+  field.addEventListener('input', myLocalStorage);
+})
+
+// execute this function once event is fired
+
+function myLocalStorage(e) {
+  formInputData[e.target.name] = e.target.value;
+    localStorage.setItem('formInputData', JSON.stringify('formInputData'));
+}
```

Comment:
- [ ] Kindly make sure the contact form information is persistent to page reloads.
- [ ] And take in considerations after filling the form, and not submitting it, the entered values should be saved in the browser.
- [ ] Consider also when visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page